frontend/src/bundles/common/components/sidebar/sidebar.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+    useEffect,
+    useLocation,
+    useState,
+} from '~/bundles/common/hooks/hooks.js';
+
+import { AppRoute } from '../../enums/app-route.enum.js';
+import { getValidClassNames } from '../../helpers/helpers.js';
+import { Icon } from '../components.js';
+import {
+    HelpIcon,
+    LogoutIcon,
+    OverviewIcon,
+    ScheduleIcon,
+} from '../icon/enums/icon-components.enum.js';
+import { sizeToClass } from '../icon/icon.js';
+import { SidebarNav } from './components/sidebar-nav/sidebar-nav.js';
+import { iconsSvgr } from './enums/enums.js';
+
+type Properties = {
+    isOpen?: boolean;
+};
+
+const styles = {
+    baseStyle:
+        'bg-lm-black-100 flex h-[95vh] w-72 flex-col content-center items-center p-7 text-white',
+    animationStyle: 'transition-transform duration-[0.5s] ease-[ease-in-out]',
+};
+
+const Sidebar = ({ isOpen = true }: Properties): JSX.Element => {
+    const { pathname } = useLocation();
+
+    const [activeRoute, setActiveRoute] = useState(pathname);
+
+    const [sidebarStyle, setSidebarStyle] = useState({});
+    useEffect(() => {
+        setActiveRoute(pathname);
+    }, [pathname]);
+
+    useEffect(() => {
+        isOpen
+            ? setSidebarStyle({ transform: 'translateX(0)' })
+            : setSidebarStyle({ transform: 'translateX(-100%)' });
+    }, [isOpen]);
+
+    return (
+        <div
+            className={getValidClassNames(
+                styles.baseStyle,
+                styles.animationStyle,
+            )}
+            style={sidebarStyle}
+        >
+            <div className="inner h-3/4 w-full border-gray-700 ">
+                <div className="flex flex-col gap-4">
+                    <SidebarNav
+                        icon={<OverviewIcon className={sizeToClass['lg']} />}
+                        text="Overview"
+                        to={AppRoute.OVERVIEW}
+                        isActive={activeRoute === AppRoute.OVERVIEW}
+                    />
+                    <SidebarNav
+                        icon={<Icon name={iconsSvgr.WORKOUT} size={'lg'} />}
```

Comment:
use enum values instead of literals where enums is available, e.g. for size we have enum